src/source-updater.js

```diff
@@ -24,54 +24,136 @@ const actions = {
   },
   callback: (callback) => (type, sourceUpdater) => {
     callback();
+  },
+  duration: (duration) => (sourceUpdater) => {
+    sourceUpdater.mediaSource.duration = duration;
   }
 };
 
 const updating = (type, sourceUpdater) => {
   const sourceBuffer = sourceUpdater[`${type}Buffer`];
 
-  return (sourceBuffer && sourceBuffer.updating) || sourceUpdater.queue[type].pending;
+  return (sourceBuffer && sourceBuffer.updating) || sourceUpdater.queuePending[type];
```

Comment:
this will return `true` if a media source action is in process right? If not, do we want it to?

Reply:
We shouldn't get to that scenario since media source actions are processed synchronously.